tests/azure/nic_test.go

```diff
@@ -0,0 +1,69 @@
+package azure_test
+
+import (
+	"fmt"
+	"math/rand"
+	"testing"
+
+	"github.com/NeowayLabs/klb/tests/lib/azure"
+	"github.com/NeowayLabs/klb/tests/lib/azure/fixture"
+)
+
+func genNicName() string {
+	return fmt.Sprintf("klb-nic-tests-%d", rand.Intn(1000))
+}
+
+func testNicCreate(t *testing.T, f fixture.F) {
+	vnet := genVnetName()
+	nsg := genNsgName()
+	nic := genNicName()
+	subnet := genSubnetName()
+	vnetAddress := "10.116.0.0/16"
+	addrnic := "10.116.1.100"
+	subnetAddress := "10.116.1.0/24"
+
+	f.Shell.Run(
+		"./testdata/create_nsg.sh",
+		nsg,
+		f.ResGroupName,
+		f.Location,
+	)
+	_ = azure.NewNsg(f)
+
+	f.Shell.Run(
+		"./testdata/create_vnet.sh",
+		vnet,
+		f.ResGroupName,
+		f.Location,
+		vnetAddress,
+	)
+	_ = azure.NewVnet(f)
+
+	f.Shell.Run(
+		"./testdata/create_subnet.sh",
+		subnet,
+		f.ResGroupName,
+		vnet,
+		subnetAddress,
+		nsg,
+	)
+	_ = azure.NewSubnet(f)
+
+	f.Shell.Run(
+		"./testdata/create_nic.sh",
+		f.ResGroupName,
+		nic,
+		f.Location,
+		vnet,
+		subnet,
+		addrnic,
+	)
+	nics := azure.NewNic(f)
+
+	nics.AssertExists(t, nic, nsg, addrnic)
```

Comment:
I didn't check the other things because if the other resources like vnet and subnet are wrong the nic won't be created. In the sample above I set the address to IP "10.116.3.100" and the subnet is "10.116.1.0/24":


11:39:48 info:    Executing command network nic create
11:39:48 info:    Looking up the network interface "klb-nic-tests-30"
11:39:49 info:    Looking up the subnet "klb-subnet-tests-91"
11:39:50 info:    Creating network interface "klb-nic-tests-30"
11:39:55 error:   Private static IP address 10.116.3.100 does not belong to the range of subnet prefix 10.116.1.0/24.
11:39:55 error:   Error information has been recorded to /home/matheus/.azure/azure.err
11:39:55 error:   network nic create command failed